### PR TITLE
TicTacToe

### PR DESCRIPTION
**Pull request by chee-yang**
TicTacMe

### CODE REVIEW

**Comment by chee-yang** [comment]
Part2

File: src/edu/htc/tictactoe/Strategy/RandomMoveStrategy.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    public int getBestMove(){
+        return 0;
+    }
+
+    public  int getRandomMove(){
+
+
+
+        ArrayList<Random> Random  = new ArrayList<Random>();
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/Strategy/RandomMoveStrategy.java]: This "Random" variable is not needed.

File: src/edu/htc/tictactoe/Strategy/RandomMoveStrategy.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        int[] randomSquare = game.getOpenSquares();
+
+        int getRandomMove = random.nextInt(randomSquare.length + 1);
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/Strategy/RandomMoveStrategy.java]: You don't need a +1 here.  You want a random number that is between 0 and the length of your randomSquare array.

This random number isn't the random move itself, it is the position in the array.

File: src/edu/htc/tictactoe/Strategy/RandomMoveStrategy.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        return getRandomMove;
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/Strategy/RandomMoveStrategy.java]: Here you would return the value at that position in the randomSquare array
randomSquare[getRandomMove]

File: src/edu/htc/tictactoe/TicTacToe.java [hunk 4]
```diff
@@ -1,9 +1,181 @@
...
 public class TicTacToe {
 
-  public void playGame() {
-    
-  }
+    Player player1;
+    Player player2;
+    GameBoard game;
+
+    RandomMoveStrategy randomMoveStrategy = new RandomMoveStrategy(game);
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/TicTacToe.java]: The strategies should not be instance variables in the class.  They should be created only as needed in the playGame method.

File: src/edu/htc/tictactoe/TicTacToe.java [hunk 5]
```diff
@@ -1,9 +1,181 @@
...
+
+    TicTacToeStrategy strategy = new TicTacToeStrategy(game) {
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/TicTacToe.java]: You might use this in the TicTacToeStrategy class for testing in the main method, but there is no reason to have this here.

File: src/edu/htc/tictactoe/TicTacToe.java [hunk 6]
```diff
@@ -1,9 +1,181 @@
...
+            return 0;
+        }
+    };
+
+
+
+
+
+
+    Scanner scanner = new Scanner(System.in);
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/TicTacToe.java]: Make sure to keep all your instance variables private for encapsulation.

File: src/edu/htc/tictactoe/TicTacToe.java [hunk 7]
```diff
@@ -1,9 +1,181 @@
...
+                    System.out.println("Updated Board");
+
+                }
+            }
+        }
+
+        else
+        //////////////////////////////1 PLAYER GAME WITH COMPUTER/////////////////////////////
+        {
+            System.out.print("Player1 Name: ");
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/TicTacToe.java]: There is a lot of similarity between the 1 player and 2 player game logic.  See if you can't remove at least some of the code duplication.

File: src/edu/htc/tictactoe/player/HumanPlayer.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    }
+    Scanner humanPlayer = new Scanner(System.in);
+
+    public int getMove()
+
+    {
+        System.out.print(getName() + ", which square would you like to pick?");
+        String getMoveInt = humanPlayer.next();
+        Integer getMove = Integer.parseInt(getMoveInt);
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/player/HumanPlayer.java]: Make sure to add the exception handling for bad (text) input.

File: src/edu/htc/tictactoe/strategy/GoForWinStrategy.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        super(game);
+    }
+
+    @Override
+    public int getBestMove() {
+        return goForWin();
+    }
+
+    public int goForWin()
```
Comment (mbMosman) [inline src/edu/htc/tictactoe/strategy/GoForWinStrategy.java]: There is code duplication between the BlockWin and GoForWin strategies.  See if you can't combine that into a single method by making the marker a variable.

**Comment by mbMosman** [comment]
Good job so far, but try to reduce the code duplication and make sure to test for errors.

**Pull request closed without merging**